smac/runhistory/runhistory.py

```diff
@@ -335,11 +335,16 @@ def save_json(self, fn: str="runhistory.json", save_external: bool=False):
                               conf.origin is not None)}
 
         # lock `fn` (writing)
-        with filelock.FileLock(fn + ".lock"):
+        lock_file_name = fn + ".lock"
+        with filelock.SoftFileLock(lock_file_name):
             with open(fn, "w") as fp:
                 json.dump({"data": data,
                            "config_origins": config_origins,
                            "configs": configs}, fp, cls=EnumEncoder, indent=2)
+        try:
```

Comment:
This should not be necessary if using a SoftFileLock

Reply:
It should not be necessary, but it is. Else, some test cases get stuck. I did not investigate this issue but simply fixed it by introducing the try/catch blocks.